sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -69,7 +69,14 @@ private[sql] class HiveSessionCatalog(
     // Current thread context classloader may not be the one loaded the class. Need to switch
     // context classloader to initialize instance properly.
     Utils.withContextClassLoader(clazz.getClassLoader) {
-      Try(super.makeFunctionExpression(name, clazz, input)).getOrElse {
+      val t = Try(super.makeFunctionExpression(name, clazz, input))
+      if (t.isSuccess) {
+        t.asInstanceOf[Success[Expression]].get
+      } else {
+        val exception = t.asInstanceOf[Failure[Expression]].exception
+        if (exception.getMessage.contains("Invalid number of arguments for function")) {
```

Comment:
can we use a special exception type instead of relying on error message?

Reply:
> can we use a special exception type instead of relying on error message?

Add a special exception for function invalid argument , this action implement form AnalysisException, finally it will be wrapperd as AnalysisException by method `withPosition()` and throw still as `AnalysisException`.